packages/enhanced/src/lib/container/runtime/CustomRuntimePlugin.ts

```diff
@@ -0,0 +1,240 @@
+import { normalizeWebpackPath } from '@module-federation/sdk/normalize-webpack-path';
+import CustomRuntimeModule from './CustomRuntimeModule';
+const { RuntimeGlobals } = require(
+  normalizeWebpackPath('webpack'),
+) as typeof import('webpack');
+import type { Compiler, Compilation, Chunk, Module, ChunkGraph } from 'webpack';
+import { getFederationGlobalScope } from './utils';
+import fs from 'fs';
+import path from 'path';
+
+const onceForCompilationMap = new WeakMap();
+const federationGlobal = getFederationGlobalScope(RuntimeGlobals);
+import { ConcatSource } from 'webpack-sources';
+
+class RuntimeModuleChunkPlugin {
+  apply(compiler: Compiler): void {
+    compiler.hooks.thisCompilation.tap(
+      'ModuleChunkFormatPlugin',
+      (compilation: Compilation) => {
+        compilation.hooks.optimizeModuleIds.tap(
+          'ModuleChunkFormatPlugin',
+          (modules: Iterable<Module>) => {
+            for (const module of modules) {
+              const moduleId = compilation.chunkGraph.getModuleId(module);
+              if (typeof moduleId === 'string') {
+                compilation.chunkGraph.setModuleId(
+                  module,
+                  `(embed)${moduleId}`,
+                );
+              } else {
+                compilation.chunkGraph.setModuleId(module, `1000${moduleId}`);
+              }
+            }
+          },
+        );
+
+        const hooks =
+          compiler.webpack.javascript.JavascriptModulesPlugin.getCompilationHooks(
+            compilation,
+          );
+
+        hooks.renderChunk.tap(
+          'ModuleChunkFormatPlugin',
+          (
+            modules: any,
+            renderContext: { chunk: Chunk; chunkGraph: ChunkGraph },
+          ) => {
+            const { chunk, chunkGraph } = renderContext;
+
+            const source = new ConcatSource();
+            source.add('var federation = ');
+            source.add(modules);
+            source.add('\n');
+            const entries = Array.from(
+              chunkGraph.getChunkEntryModulesWithChunkGroupIterable(chunk),
+            );
+            for (let i = 0; i < entries.length; i++) {
+              const [module, entrypoint] = entries[i];
+              const final = i + 1 === entries.length;
+              const moduleId = chunkGraph.getModuleId(module);
+              source.add('\n');
+              if (final) {
+                source.add('for (var mod in federation) {\n');
```

Comment:
installs the modules into __webpack_require__.m, then calls require from parent runtime on it to install the runtime